drivers/timer/cortex_m_systick.c

```diff
@@ -432,15 +432,18 @@ void sys_clock_disable(void)
 
 static int sys_clock_driver_init(void)
 {
+	uint32_t ctrl = SysTick_CTRL_ENABLE_Msk |
+			SysTick_CTRL_TICKINT_Msk;
+#ifndef CONFIG_CORTEX_M_SYSTICK_EXT_REF_CLOCK
+	ctrl |= SysTick_CTRL_CLKSOURCE_Msk;
+#endif
 
 	NVIC_SetPriority(SysTick_IRQn, _IRQ_PRIO_OFFSET);
 	last_load = CYC_PER_TICK;
 	overflow_cyc = 0U;
 	SysTick->LOAD = last_load - 1;
 	SysTick->VAL = 0; /* resets timer to last_load */
-	SysTick->CTRL |= (SysTick_CTRL_ENABLE_Msk |
-			  SysTick_CTRL_TICKINT_Msk |
-			  SysTick_CTRL_CLKSOURCE_Msk);
+	SysTick->CTRL = ctrl;
```

Comment:
Note that I modified this from `|=` to `=` which I believe should not impact anything.
The only other bit in this register is the readonly `COUNTFLAG`:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/124088381/9fbb1c50-9007-48ce-adcc-a9c765761b94)
and CMSIS `SysTick_Config()` uses  `=`:
https://github.com/zephyrproject-rtos/cmsis/blob/master/CMSIS/Core/Include/core_cm4.h#L2032